lib/_http_server.js

```diff
@@ -503,7 +503,7 @@ function Server(options, requestListener) {
   if (typeof options === 'function') {
     requestListener = options;
     options = {};
-  } else if (options == null || typeof options === 'object') {
+  } else if (options == null || (typeof options === 'object' && !ArrayIsArray(options))) {
     options = { ...options };
   } else {
     throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
```

Comment:
IMO a better fix would be:
```suggestion
    options = kEmptyObject;
  } else if (options == null) {
    options = kEmptyObject;
  } else {
    validateObject(options, 'options');
```

Reply:
We can't use kEmptyobject for options now because options is mutated in later code. But rest of your suggestion is applied. Thank you for enhancement.

Reply:
That's a problem, in this case we cannot pass the same object as the one passed by users. We need to either get rid of the mutation, or keep the previous version.